app-creator/client/src/widgets/ui-sidemenu/ui-sidemenu.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ *  @fileoverview The ui-sidemenu is used as a wrapper for the different tabs
+ *  in the actions-panel.
+ */
+import {
+  LitElement,
+  html,
+  customElement,
+  css,
+  property,
+  query,
+} from 'lit-element';
+import { Panel } from '../ui-panel/ui-panel';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+import '@polymer/iron-icons/iron-icons.js';
+import '../ui-panel/ui-panel';
+
+@customElement('ui-sidemenu')
+export class SideMenu extends LitElement {
+  /**
+   * Additional custom styles.
+   */
+  static styles = css`
+    paper-icon-button {
+      margin: var(--tight);
+      pointer-events: auto;
+      background-color: white;
+      border-radius: 50%;
+    }
+
+    #container {
+      display: flex;
+      height: 100%;
+      top: 0;
+      left: 0;
+      position: absolute;
+      width: 100%;
+      pointer-events: none;
+    }
+
+    ui-panel {
+      height: 100%;
+      width: 80%;
+      overflow: hidden;
+      background-color: var(--primary-color);
+      pointer-events: auto;
+    }
+
+    .full-height {
+      height: 100%;
+    }
+
+    .column {
+      flex-direction: column;
+    }
+
+    .row {
+      flex-direction: row;
+    }
+  `;
+
+  /**
+   * Sets the flex layout of child widgets.
+   * Options available are 'column' and 'row'.
+   * column layout will append widgets below the last child element.
+   * row layout will append widgets to the right of the last child element.
+   */
+  @property({ type: String }) layout = 'column';
+
+  /**
+   * Contains an inner dropzone-widget.
+   */
+  @property({ type: Boolean }) hasDropzone = false;
+
+  /**
+   * Additional custom styles
+   */
+  @property({ type: Object }) styles = {};
+
+  /**
+   * Sets editable property.
+   */
+  @property({ type: Boolean }) editable = false;
+
+  /**
+   * Reference to ui panel widget.
+   */
+  @query('ui-panel') panel!: Panel;
+
+  firstUpdated() {
+    if (this.panel != null) {
+      this.panel.id = this.id;
+    }
+  }
+
+  toggleMenu(e: Event) {
+    e.stopPropagation();
+
+    const { panel } = this;
+    if (panel == null) {
```

Comment:
`if (!panel)` covers `undefined` as well, though I can't imagine when or how `this` can be undefined...

Reply:
Double equals`==` cover undefined as well, so if panel is null or undefined, it will return true. If we had `===` instead then it would not cover undefined. 

I use `if (panel == null)` instead of `if (!panel)` just to be more explicit of what we are expecting panel to be. It might be unnecessary for this case.